src/callbacks.jl

```diff
@@ -203,9 +203,15 @@ function apply_callback!(integrator,callback::ContinuousCallback,cb_time,prev_si
     change_t_via_interpolation!(integrator,integrator.tprev+cb_time)
   end
   saved_in_cb = false
+  oldopt = integrator.opts.save_everystep
 
   @inbounds if callback.save_positions[1]
```

Comment:
This still only runs when `callback.save_positions[1] == true`, so the `saveat`s are skipped. I would do a normal `savevalues!` before checking this, and then check to whether you should force one additional save by checking `sol.t` for if it was saved already.